geopandas/io/sql.py

```diff
@@ -12,35 +12,50 @@ def read_postgis(sql, con, geom_col='geom', crs=None, index_col=None,
     Returns a GeoDataFrame corresponding to the result of the query
     string, which must contain a geometry column.
 
-    Examples:
-    sql = "SELECT geom, kind FROM polygons;"
-    df = geopandas.read_postgis(sql, con)
-
     Parameters
     ----------
-    sql: string
-    con: DB connection object or SQLAlchemy engine
-    geom_col: string, default 'geom'
+    sql : string
+        SQL query to execute in selecting entries from database.
+    con : DB connection object or SQLAlchemy engine
+        Active connection to the database to query.
+    geom_col : string, default 'geom'
         column name to convert to shapely geometries
-    crs: optional
+    crs : dict or str, optional
         CRS to use for the returned GeoDataFrame
 
     See the documentation for pandas.read_sql for further explanation
     of the following parameters:
     index_col, coerce_float, params
 
+    Returns
+    -------
+    geodataframe : GeoDataFrame
+
+    Example
+    -------
+    >>> sql = "SELECT geom, kind FROM polygons;"
+    >>> df = geopandas.read_postgis(sql, con)
     """
-    df = read_sql(sql, con, index_col=index_col, coerce_float=coerce_float,
-                  params=params)
 
-    if geom_col not in df:
-        raise ValueError("Query missing geometry column '{0}'".format(
-            geom_col))
```

Comment:
If there's no `geom_col` in the database, it'll now error in the `read_sql` call. The error message there seems informative enough, so didn't do anything to catch and explain it.